src/Dijkstra/AST/Examples/Maybe/Bind.agda

```diff
@@ -0,0 +1,106 @@
+open import Data.Nat renaming (ℕ to Nat)
+open import Dijkstra.AST.Core
+open import Dijkstra.AST.Maybe
+open        MaybeSyntax
+open        ASTOpSem         MaybeOpSem
+open        ASTPredTrans     MaybePT
+open        ASTPredTransMono MaybePTMono
+open        ASTSufficientPT  MaybeSuf
+open        ASTTypes         MaybeTypes
+open import Haskell.Prelude
+open import Util.Prelude
+
+module Dijkstra.AST.Examples.Maybe.Bind (mn1 mn2 : MaybeD Nat) where
+
+module OneMaybeBindExample where
+  prog : MaybeD (List Nat)
+  prog = do
+    n1 <- mn1
+    return (n1 ∷ [])
+
+  ProgPost : Maybe (List Nat) -> Set
+  ProgPost nothing = ⊤
+  ProgPost (just l) = length l ≡ 1
+
+  mn1Post : Post Nat
+  mn1Post nothing = ⊤
+  mn1Post (just n) = runMaybe mn1 unit ≡ just n
+
+  -- Here is the property we want to prove
+  progPostWP : predTrans prog ProgPost unit
+  -- This long-winded proof was helpful in understanding how to make the proof work
+  -- Agda knows the Goal postcondition because it knows that prog is a bind, and knows the rest of
+  -- the program.  To help us understand what it is that Agda figures out to enable putting _ for
+  -- the goal argument below, we define Goal below, and we can replace _ by Goal and see that it's
+  -- right.
+  progPostWP = predTransMono mn1 mn1Post _ {- Goal -} mn1Post⇒Goal unit PT
+    where
+
+    Goal : Post Nat
+    Goal x = -- bindPT (λ x → predTrans (Monad.return MonadAST (x ∷ []))) unit ProgPost
```

Comment:
perhaps add a comment about what this comment means

Reply:
DONE